### ISSUE TITLE
Super Fast Scroll Speed

### ISSUE THREAD

**Issue by dragonmacher** [JFormDesigner/FlatLaf#892]
On the Ghidra project, we have noticed users complaining of fast scroll speed while scrolling the mouse inside of scroll panes that contain JEditorPanes.  Scrolling 1 tick at a time using the mouse wheel works as expected.  Scrolling faster than that causes the entire view to jump to the bottom.   I believe the larger the content of the text pane, the faster the scrolling. 

It seems like the issue is in the method `mouseWheelMovedSmooth()` is incorrectly calculating the height to scroll, at the line below.   The BasicScrollPaneUI uses this calculation, which seems to work well:

```
Scrollable scrollComp = (Scrollable) comp;
Rectangle viewRect = vp.getViewRect();
```

https://github.com/JFormDesigner/FlatLaf/blob/4e6f5385194685296b542269461c94fbad9ba8ca/flatlaf-core/src/main/java/com/formdev/flatlaf/ui/FlatScrollPaneUI.java#L213

We are using an older release of FlatLaf (3.2.1), but the code appears unchanged.

For now, we have disabled this behavior using this code:

```
UIManager.put("ScrollPane.smoothScrolling", Boolean.FALSE);	
```

FlatLaf 3.2.1
Java 21.0.3
Mac Sonoma 14.6.1

**Comment by DevCharly** [comment]
related Ghidra issue: https://github.com/NationalSecurityAgency/ghidra/issues/6952

Thanks for reporting.
You're right.
FlatLaf passes the wrong (too large) height into `Scrollable.getScrollableUnitIncrement( visibleRect, ... )`.
And scrolling is too fast because `JTextComponent.getScrollableUnitIncrement(...)`
is implemented as `return visibleRect.height / 10;`.

Seems that too fast scrolling is only a problem in text components.

JTree, JTable and JList are not affected because they do not use 
`visibleRect.height` in their implementations of `getScrollableUnitIncrement(...)`.

**Comment by DevCharly** [comment]
fixed in latest `3.5.2-SNAPSHOT`: https://github.com/JFormDesigner/FlatLaf#snapshots
